### PR TITLE
Update the Readme for the lib-megashark

### LINKED ISSUE #13: Explain how to use, build, install in README

### PR DESCRIPTION
**Pull request by fabienSvtr**
closes #13

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,73 @@
-Generic UI library based on Ionic
+# Generic UI library based on Ionic
+
+The megashark-lib library is a UI component library. It is used to centralise all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).
```
Comment (Ironicbay) [inline README.md:3]:
```suggestion
The megashark-lib library is a UI component library. It is used to centralize all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).
```

File: README.md [hunk 2]
```diff
@@ -1 +1,73 @@
...
+
+```
+git clone git@github.com:Scille/megashark-lib.git
+```
+
+## 📁 Folder names
+
+**src** (local app): folder used to test the application locally. It includes a component display page.
+
+**lib**: includes all the libraryce components that you want to have in common in the library.
```
Comment (Ironicbay) [inline README.md:19]: `libraryce` ?

File: README.md [hunk 3]
```diff
@@ -1 +1,73 @@
...
+
+## 🧑‍💻 Use in the development space
+
+### Modify a megashark-lib component and see the result
+
+> In the **megashark-lib**:
```
Comment (Ironicbay) [inline README.md:25]:
```suggestion
> In **megashark-lib**:
```

File: README.md [hunk 4]
```diff
@@ -1 +1,73 @@
...
+
+When you push on `master` → This creates a new version of the library.
+
+### Modify a component of **megashark-lib** at the same time as using it in **parsec-cloud**
+
+This allows you to see changes made to the library and see them displayed in real time in the `parsec-cloud` project.  Please follow all the commands in the order below.
+
+1️⃣ **Build the project and install it globally**
+
+> In the **megashark-lib**
```
Comment (Ironicbay) [inline README.md:38]:
```suggestion
> In **megashark-lib**
```

File: README.md [hunk 5]
```diff
@@ -1 +1,73 @@
...
+```
+
+The command includes (not to be written in the terminal) :
+
+`npm link`: to create a link to another project from a local environment.
+`vite build --watch`: to rebuild automatically as soon as there is a modification.
+
+2️⃣ **Link the libraries from parsec-cloud**.
+
+> In the **parsec-cloud**
```
Comment (Ironicbay) [inline README.md:50]:
```suggestion
> In **parsec-cloud**
```

File: README.md [hunk 6]
```diff
@@ -1 +1,73 @@
...
+```
+npm run lib:local:start
+```
+
+The command includes (not to be written in the terminal):
+`npm link lib-megashark`: allows you to link the lib-megashark library (installed from the machine to github)
+
+3️⃣ **Start development**
+> In the **parsec-cloud**
```
Comment (Ironicbay) [inline README.md:59]:
```suggestion
> In **parsec-cloud**
```

File: README.md [hunk 7]
```diff
@@ -1 +1,73 @@
...
+```
+npm run web:open //web
+npm run electron:open //electron
+```
+
+4️⃣ **Reset installation to state (with remote project)**
+
+⚠️ Required to keep the project with the remote link.
+> In the **parsec-cloud**
```
Comment (Ironicbay) [inline README.md:68]:
```suggestion
> In **parsec-cloud**
```

**Review by Ironicbay — CHANGES_REQUESTED** [review]

**Review by Ironicbay — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -1 +1,80 @@
...
+
+The command includes (not to be written in the terminal):
+`npm link lib-megashark`: allows you to link the lib-megashark library (installed from the machine to github)
+
+3️⃣ **Start development**
+
+> In **parsec-cloud**
+
+```
+npm run web:open //web
```
Comment (Max-7) [inline README.md:66]: Comments use `#`, not `//`

File: README.md [hunk 9]
```diff
@@ -1 +1,80 @@
...
+```
+
+4️⃣ **Reset installation to state (with remote project)**
+
+⚠️ Required to keep the project with the remote link.
+
+> In **parsec-cloud**
+
+```
+npm run lib:local:stop
```
Comment (Max-7) [inline README.md:77]: Renamed to `megashark:local:stop`

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -1 +1,80 @@
-Generic UI library based on Ionic
+# Generic UI library based on Ionic
+
+The megashark-lib library is a UI component library. It is used to centralize all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).
```
Comment (NicoTuxx) [inline README.md:3]:
```suggestion
The megashark-lib is a UI component library. It is used to centralize all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).
```

**Review by NicoTuxx — CHANGES_REQUESTED** [review]

File: README.md [hunk 11]
```diff
@@ -1 +1,80 @@
...
+
+## 🏁 Installation
+
+### Clone the local directory
+
+You will need to clone the library directory locally using git.
+
+```
+git clone git@github.com:Scille/megashark-lib.git
+```
```
Comment (NicoTuxx) [inline README.md:13]: Installation needs npm install too

File: README.md [hunk 12]
```diff
@@ -1 +1,80 @@
...
+
+```
+npm run web:open
+```
+
+When you push on `master` → This creates a new version of the library.
+
+### Modify a component of **megashark-lib** at the same time as using it in **parsec-cloud**
+
+This allows you to see changes made to the library and see them displayed in real time in the `parsec-cloud` project. Please follow all the commands in the order below.
```
Comment (NicoTuxx) [inline README.md:35]:
```suggestion
This allows you to see changes made to the library and see them displayed in real time in the project using this library. Please follow all the commands in the order below.
```

**Review by NicoTuxx — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -1 +1,80 @@
...
+
+### Modify a megashark-lib component and see the result
+
+> In **megashark-lib**:
+
+```
+npm run web:open
+```
+
+When you push on `master` → This creates a new version of the library.
```
[the remark this answers is not in the document]
Comment (Max-7) [reply README.md:31]: Not really, resyncs should not be done all the time, they should be planned, with a tag and everything. The process of creating a tag (or a "new version") does not need to be detailled, it's up to the maintainer.

File: README.md [hunk 14]
```diff
@@ -1 +1,99 @@
...
+### 2️⃣ Link the libraries from your project
+
+> In **your project**
+
+```
+npm run megashark:local:start
+```
+
+The command includes (not to be written in the terminal):
+`npm link lib-megashark`: allows you to link the lib-megashark library (installed from the machine to github)
```
Comment (NicoTuxx) [inline README.md:75]:
```suggestion
`npm link megashark-lib`: allows you to link the megashark-lib library (installed from the machine to github)
```

File: README.md [hunk 15]
```diff
@@ -1 +1,99 @@
...
+
+### 3️⃣ Start development
+
+> In **your project**
+
+```
+# For web project
+npm run web:open#web
```
Comment (NicoTuxx) [inline README.md:83]:
```suggestion
npm run web:open
```

**Review by NicoTuxx — CHANGES_REQUESTED** [review]

**Review by NicoTuxx — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]

**Pull request merged**